conduit/pipeline.go

```diff
@@ -118,13 +112,17 @@ func MakePipelineConfig(logger *log.Logger, cfg *Config) (*PipelineConfig, error
 // exporters and generating the result
 type Pipeline interface {
 	Start() error
-	Stop() error
+	Stop()
+	Error() error
 }
 
 type pipelineImpl struct {
-	ctx    context.Context
-	cfg    *PipelineConfig
-	logger *log.Logger
+	ctx     context.Context
+	cf      context.CancelFunc
+	running sync.WaitGroup
```

Comment:
nit: consider naming this wg so it's clear that it's a go waitgroup.

Reply:
I've copied the name from https://github.com/algorand/go-algorand/blob/master/catchup/catchpointService.go#L75 since I liked it.

To me, `running is a waitgroup counting the running goroutine(1)` makes sense. Open to changing it if we want to stick with `wg` for waitgroup vars though.

Reply:
I agree with Shiqi here. Though I've never seen it written down, calling this `wg` seems to be a go idiom similar to `ctx` and `cf` above.